cpp/src/arrow/compute/exec/expression.cc

```diff
@@ -634,6 +634,20 @@ std::optional<Out> FoldLeft(It begin, It end, const BinOp& bin_op) {
   return folded;
 }
 
+template <typename BinOp, typename It,
+          typename Value = typename std::iterator_traits<It>::value_type,
+          typename Out = decltype(std::declval<BinOp>()(std::declval<Value>(),
+                                                        std::declval<Value>()))>
+std::optional<Out> MaybeFoldLeft(It begin, It end, const BinOp& bin_op) {
```

Comment:
Nit: Is this the same thing as `std::accumulate`?  Could we maybe use `MaybeAccumulate` or, in a brief comment, reference `std::accumulate` for new readers who might not be familiar with the term "folding" (I often think of this as "reduce" but I just looked it up and it seems there is a `std::reduce` but it doesn't guarantee **left**-folding and thus can run in parallel)

Reply:
It is the same as std::accumulate. I'll just remove these helpers; they don't add much